worker/uniter/runner/jujuc/network-get_test.go

```diff
@@ -62,32 +112,72 @@ func (s *NetworkGetSuite) TestNetworkGet(c *gc.C) {
 		args:    []string{""},
 		out:     `no binding name specified`,
 	}, {
-		summary: "binding name given, no --primary-address given",
-		code:    2,
-		args:    []string{"foo"},
-		out:     `--primary-address is currently required`,
+		summary: "unknown binding given, no --primary-address given",
+		code:    1,
+		args:    []string{"unknown"},
+		out:     `no network config found for binding "unknown"`,
 	}, {
 		summary: "unknown binding given, with --primary-address",
 		args:    []string{"unknown", "--primary-address"},
 		code:    1,
-		out:     "insert server error for unknown binding here",
+		out:     `no network config found for binding "unknown"`,
 	}, {
-		summary: "valid arguments, API server returns no config",
+		summary: "API server returns no config for this binding, with --primary-address",
 		args:    []string{"valid-no-config", "--primary-address"},
 		code:    1,
 		out:     `no network config found for binding "valid-no-config"`,
 	}, {
+		summary: "API server returns no config for this binding, no --primary-address",
+		args:    []string{"valid-no-config"},
+		code:    1,
+		out:     `no network config found for binding "valid-no-config"`,
+	}, {
 		summary: "explicitly bound, extra-binding name given with --primary-address",
 		args:    []string{"known-extra", "--primary-address"},
 		out:     "10.20.1.42",
 	}, {
+		summary: "explicitly bound, extra-binding name given without --primary-address",
+		args:    []string{"known-extra"},
+		out: `- macaddress: "00:11:22:33:44:22"
```

Comment:
Done, although I don't know if that's the right way to do it. Shouldn't 'omitempty' for error in NetworkInfoResult cause it not to be printed if it's nil?